doc/hacking.rst

```diff
@@ -531,6 +531,10 @@ Good:
 If ``bar`` and ``baz`` are already public, then there's no need to add yet
 another helper to save the caller a line of code.
 
+- Prefer ``open`` over ``open!``. The ``open!`` directive suppresses unused open
+  warnings, which can lead to pointless dependencies. Only use ``open!`` when the
+  open statement is genuinely unused but required for other reasons.
```

Comment:
I don't think people should spending their brain cycles on this. The downsides aren't significant enough. Feel free to clean this up using manual/automated ways though.